### PR TITLE
lab/memory-gdb: Add comment about difference between char *s and char s[]

### LINKED ISSUE #122: lab/memory-gdb: Add comment about difference betweehn `char *s` and `char s[]`
[This line](https://github.com/systems-cs-pub-ro/iocla/blob/56d3dccb47fea0f801281462b54e65aa0a944b08/laborator/content/operatii-memorie-gdb/2-delete-first/delete-first.c#L8) needs to be modified to `char s[] = "Ana are mere";` for the exercise to function properly. This is intentional, but students often miss [this hint](https://github.com/systems-cs-pub-ro/iocla/blob/56d3dccb47fea0f801281462b54e65aa0a944b08/laborator/content/operatii-memorie-gdb/README.md?plain=1#L400-L404) in the lab text. A `TODO` comment should be added before the aforementioned code to ask the students if that declaration is correct and why.

### PR DESCRIPTION
**Pull request by dumitrache-adrian92**
Added a comment hinting at a possible mistake in the declaration of s. 
Fixes #122

### CODE REVIEW

File: laborator/content/operatii-memorie-gdb/2-delete-first/delete-first.c [hunk 1]
```diff
@@ -5,6 +5,10 @@
 char* delete_first(char *s, char *pattern);
 
 int main(){
```
Comment (teodutu) [inline laborator/content/operatii-memorie-gdb/2-delete-first/delete-first.c:7]:
```suggestion
int main()
{
```

File: laborator/content/operatii-memorie-gdb/2-delete-first/delete-first.c [hunk 2]
```diff
@@ -5,6 +5,10 @@
...
+	/*
+	TODO: Este corecta declaratia variabilei s in contextul in care o sa apelam
```
Comment (teodutu) [inline laborator/content/operatii-memorie-gdb/2-delete-first/delete-first.c:9]: Use diacritics and mind the `checkpatch.pl` error below. Comments should look like this:

```c
/**
 * This is a comment.
 */
```

**Review by teodutu — CHANGES_REQUESTED** [review]
The content of the comment is good. You only have some cosmetic changes to make.

**Comment by dumitrache-adrian92** [comment]
Sorry for the double commit, I made a tiny mistake. I added diacritics, fixed that parenthesis and corrected the formatting.

**Review by teodutu — APPROVED** [review]
`checkpatch.pl` has found another error. It's honestly a small detail, but please fix it and then we can merge this PR.

**Pull request merged**